src/ConfigProvider/CachedConfigProvider.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Tnapf\Config\ConfigProvider;
+
+use DateInterval;
+use Psr\SimpleCache\CacheInterface;
+
+class CachedConfigProvider implements ConfigProvider
+{
+    public function __construct(
+        private readonly ConfigProvider $configProvider,
+        private readonly CacheInterface $cache,
+        private readonly DateInterval|int|null $ttl = null
+    ) {
+    }
+
+    public function get(string $key): mixed
```

Comment:
The possible `InvalidArgumentException` here should be caught, and then throw a new `InvalidConfigException` with previous set to the caught exception.